src/extension.rs

```diff
@@ -114,6 +119,25 @@ pub trait AuthClient: ExtensionClient<AuthExtension> {
         })
     }
 
+    /// Reset a pin.
+    ///
+    /// Similar to [`set_pin`](AuthClient::set_pin), but allows the key that the pin will unwrap to be configured.
+    /// This allows for example backing up the key for a pin, to be able to restore it from another source.
```

Comment:
Restrict `key` to keys returned by `get_pin_key`?

Reply:
I’m only talking about the docs.  Currently it is not clear what type of key can be used.  We should either describe the requirements or restrict it to keys returned by `get_pin_key`.

Reply:
Currently there is a check that it's correctly a `Symmetric(32)` key, I added the documentation to restrict it to keys from `get_pin_key`.